### ISSUE TITLE
tf.layers.Dense work error with unit is a Tensor

### ISSUE THREAD

**Issue by zh794390558** [tensorflow/tensorflow#16775]
OS Platform and Distribution
TensorFlow installed from
docker hub

TensorFlow version
1.5.0

CUDA/cuDNN version
CUDA9

GPU model and memory
TITAN XP

Exact command to reproduce
N/A

### Describe the problem
`tf.layers.Dense._rnn_output_size()` not work when `self.units` is a Tensor.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by zh794390558** [comment]
@tensorflowbutler fix

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by fchollet** [comment]
`units` is an `int`, as mentioned explicitly in the docs. It's not something that you can dynamically change as part of a computation graph, it's a hardcoded architecture parameter (since it determines the dimensions of the layer's weights).

Docs:

```
units: Positive integer, dimensionality of the output space.
```
